shopfloor/models/stock_move.py

```diff
@@ -36,3 +36,36 @@ def _action_done(self, cancel_backorder=False):
                 if picking.state == "done":
                     picking.action_done()
         return moves
+
+    def action_done_in_picking(self):
+        """Put the moves in a separate transfer and validate them."""
+        moves = self.filtered(lambda m: m.state == "assigned")
+        if not moves:
+            return False
+        for picking in moves.picking_id:
+            new_picking = picking.copy({
+                'name': '/',
+                'move_lines': [],
+                'move_line_ids': [],
+                'backorder_id': picking.id
+            })
+            picking.message_post(
+                body=_(
+                    'The backorder <a href=# data-oe-model=stock.picking data-oe-id=%d>'
+                    '%s</a> has been created.'
+                ) % (
+                    new_picking.id, new_picking.name
+                )
+            )
+            moves_todo = picking.move_lines & moves
+            moves_todo.write({'picking_id': new_picking.id})
+            moves_todo.mapped('package_level_id').write({'picking_id': new_picking.id})
+            moves_todo.mapped('move_line_ids').write({'picking_id': new_picking.id})
+            new_picking.action_assign()
+            assert new_picking.state == "assigned"
+            # Validate the new picking
+            for move_line in moves_todo.move_line_ids:
+                if not move_line.qty_done:
```

Comment:
Do you expect "users" of this method to split the move lines beforehand if we handle them one by one?
An alternative take would be to expect the qty_done to be already set before calling this method, and this method calling the method that split the lines without qty_done in another move.

Reply:
Yes I was expecting that, need to write a docstring about the usage